build.gradle

```diff
@@ -36,15 +36,51 @@ buildscript {
 
 repositories {
     mavenCentral()
+    maven {
+        // Confluent repo for kafka-avro-serializer
+        url "https://packages.confluent.io/maven/"
+    }
 }
 
 dependencies {
-    compile 'org.apache.kafka:kafka-clients:2.4.1'
-    compile 'com.github.luben:zstd-jni:1.4.3-1'
-    compile 'org.slf4j:slf4j-api:1.7.28'
-    compile 'org.lz4:lz4-java:1.6.0'
+    compile('io.confluent:kafka-avro-serializer:5.5.1') {
+        exclude group: 'org.apache.kafka', module:'kafka-clients'
+    }
+    compile('io.confluent:kafka-schema-serializer:5.5.1') {
+        exclude group: 'org.apache.kafka', module:'kafka-clients'
+    }
+    compile 'io.confluent:common-config:5.5.1'
+    compile 'org.apache.avro:avro:1.9.2'
+    compile('io.confluent:kafka-schema-registry-client:5.5.1') { // not open source!
+        exclude group: 'org.apache.kafka', module:'kafka-clients'
+    }
+    compile('org.apache.kafka:kafka_2.12:2.5.1') { // contains kafka.utils.VerifiableProperties used by kafka-schema-registry-client
+        exclude group: 'org.apache.zookeeper', module: 'zookeeper'
```

Comment:
Can we instead use [`include`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceTask.html#org.gradle.api.tasks.SourceTask:include(groovy.lang.Closure) to explicitly include the `kafka.utils.VerifiableProperties` that we need instead of manually excluding a bunch of unrelated things?

Reply:
That method is only for manipulation of file trees not transitive dependency management. 
To avoid transitive dependency resolution we could use `transitive = false` but also with that `kafka_2.12` transitively include group: 'org.scala-lang.modules', module:' scala-java8-compat_2.12'